src/store/metadata/oxigraph/store.py

```diff
@@ -73,31 +76,33 @@ def get_dataset(self, dataset_id: str) -> Optional[Dict]:
         #
         # The user doesnt need to know about blank RDF nodes so we need
         # to flatten and embed the latter two graphs in the dataset graph.
-        dataset_graph = next((x for x in data["@graph"] if "@type" in x.keys()), None)
-        contact_point_graph = next(
-            (x for x in data["@graph"] if "vcard:fn" in x.keys()), None
-        )
-        temporal_coverage_graph = next(
-            (x for x in data["@graph"] if "dcat:endDate" in x.keys()), None
-        )
+        dataset_graph =  _get_single_graph_for_field(data, "@type")
+        contact_point_graph = _get_single_graph_for_field(data, "vcard:fn") 
+        temporal_coverage_graph = _get_single_graph_for_field(data, "dcat:endDate")
+        
+        if None in [dataset_graph, contact_point_graph,temporal_coverage_graph]:
+            return None
+
+        # Add `issued` and `modified` fields to each version in `versions`
```

Comment:
Sorry, being really picky - this should be "Add `issued` and `modified` fields to each edition in `editions`" (your future self will thank you) - this is literally the only change